### PR TITLE
Added Popularity Contest module/service

### PR DESCRIPTION
**Pull request by jmazouri**
Added IMessageExtensions.cs
- Add GetGuild and GetGuildId extension methods
- Add GetMessageLink extension method, supplants the old method in existing code
Added IEmote type reader

Since this feature is nondestructive, but is difficult to test with hand-mocked data, my intent is to get it up and running and do testing in private channels before we actually use it.

### CODE REVIEW

File: Modix/IEmoteTypeReader.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Modix
+{
+    public class IEmoteTypeReader : TypeReader
+    {
+//TypeReaders have to be asynchronous but we don't do async work, so...
+#pragma warning disable 1998
```
Comment (JakenVeina) [inline Modix/IEmoteTypeReader.cs]: Uhh.....no? You absolutely do not need to put the `async` keyword on that method. It simply needs to return `Task<TypeReaderResult>`. Which you can do with `Task.FromResult()`.

File: Modix/IEmoteTypeReader.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
+﻿using Discord;
+using Discord.Commands;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Modix
+{
+    public class IEmoteTypeReader : TypeReader
```
Comment (JakenVeina) [inline Modix/IEmoteTypeReader.cs]: I know the intention is that "this is a `TypeReader` that reads `IEmote` values", but I think it's still better to not put `I` at the beginning of a `class` name.

File: Modix.Services/PopularityContest/PopularityContestService.cs [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+using Modix.Services.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Modix.Services.PopularityContest
+{
+    public class PopularityContestService
```
Comment (JakenVeina) [inline Modix.Services/PopularityContest/PopularityContestService.cs]: You mind throwing up an `IPopularityContestService` interface as well, for consistency and future testings?

File: Modix/IEmoteTypeReader.cs (hunk 1 above)
Comment (jmazouri) [reply Modix/IEmoteTypeReader.cs]: Gah, totally slipped my mind. Will change

**Pull request merged**